napari/utils/events/containers/_set.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+from typing import Any, Iterable, Iterator, MutableSet, TypeVar
+
+from napari.utils.events import EmitterGroup
+
+_T = TypeVar("_T")
+
+
+class EventedSet(MutableSet[_T]):
+    """An unordered collection of unique elements.
+
+    Parameters
+    ----------
+    data : iterable, optional
+        Elements to initialize the set with.
+
+    Events
+    ------
+    added (value: _T)
+        emitted after an item is added to the set.  Will not be emitted if an
+        item was already in the set when added.
+    removed (value: _T)
+        emitted after an item is removed from the set.  Will not be emitted if
+        the item was not in the set when discarded.
+    """
+
+    events: EmitterGroup
+
+    def __init__(self, data: Iterable[_T] = ()):
+
+        _events = {'added': None, 'removed': None}
+        # For inheritance: If the mro already provides an EmitterGroup, add...
+        if hasattr(self, 'events') and isinstance(self.events, EmitterGroup):
+            self.events.add(**_events)
+        else:
+            # otherwise create a new one
+            self.events = EmitterGroup(source=self, **_events)
+
+        self._set: set[_T] = set()
+        self.update(data)
+
+    # >>>>>>>> Required Abstract Methods
+
+    def __contains__(self, x: Any) -> bool:
+        return x in self._set
+
+    def __iter__(self) -> Iterator[_T]:
+        return self._set.__iter__()
+
+    def __len__(self) -> int:
+        return len(self._set)
+
+    def add(self, value: _T) -> None:
+        if value not in self:
+            self._set.add(value)
+            self.events.added(value=value)
+
+    def discard(self, value: _T) -> None:
+        """Remove an element from a set if it is a member.
+
+        If the element is not a member, do nothing.
+        """
+        if value in self:
+            self._set.discard(value)
+            self.events.removed(value=value)
+
+    # <<<<<<<< End Abstract Methods
```

Comment:
Do those marker have any change of confusing tools that deal with git conflicts ?

Reply:
oh!  they might! :joy: ... will change